Extensions/Guardrail+Settings.swift

```diff
@@ -84,58 +84,74 @@ public extension Guardrail where Value == HKQuantity {
         recommendedBounds: 16...399,
         unit: HKUnit.milligramsPerDeciliter.unitDivided(by: .internationalUnit())
     )
-
+ 
+    static let absoluteMinimumCarbRatio = 2.0
     static let carbRatio = Guardrail(
-        absoluteBounds: 1...150,
-        recommendedBounds: 3.0.nextUp...28.0.nextDown,
+        absoluteBounds: absoluteMinimumCarbRatio...150,
+        recommendedBounds: 4...28,
         unit: .gramsPerUnit
     )
 
     static func basalRate(supportedBasalRates: [Double]) -> Guardrail {
+        let scheduledBasalRateAbsoluteRange = 0.05...30.0
         let recommendedLowerBound = supportedBasalRates.first == 0
             ? supportedBasalRates.dropFirst().first!
             : supportedBasalRates.first!
         return Guardrail(
-            absoluteBounds: supportedBasalRates.first!...supportedBasalRates.last!,
-            recommendedBounds: recommendedLowerBound...supportedBasalRates.last!,
+            absoluteBounds: (supportedBasalRates.first!...supportedBasalRates.last!).clamped(to: scheduledBasalRateAbsoluteRange),
+            recommendedBounds: (recommendedLowerBound...supportedBasalRates.last!).clamped(to: scheduledBasalRateAbsoluteRange),
             unit: .internationalUnitsPerHour
         )
     }
 
-    static var recommendedMaximumScheduledBasalScaleFactor: Double {
-        return 6
-    }
-
     static func maximumBasalRate(
         supportedBasalRates: [Double],
         scheduledBasalRange: ClosedRange<Double>?,
+        lowestCarbRatio: Double?,
         maximumBasalRatePrecision decimalPlaces: Int = 3
     ) -> Guardrail {
-        let minimumSupportedBasalRate = supportedBasalRates.first!
-        let recommendedLowerBound = minimumSupportedBasalRate == 0 ? supportedBasalRates.dropFirst().first! : minimumSupportedBasalRate
+        
+        let maximum = 70.0 / (lowestCarbRatio ?? absoluteMinimumCarbRatio)
```

Comment:
[nit] Consider replacing `absoluteMinimumCarbRatio` with `carbRatio.absoluteBounds.lowerBound`? (And getting rid of the "special" `absoluteMinimumCarbRatio`.)

Reply:
I did that to avoid having to use the `unit`, but it's minor, I'll do as you suggest...